_posts/2021-04-28-cockpit-243.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Cockpit 243
+author: pitti
+date: 2021-04-28
+tags: cockpit linux
+slug: cockpit-243
+category: release
+summary: Cockpit with Services page improvements
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.
+
+Here are the release notes from Cockpit version 243.
+
+## Services: Show sockets and memory usage
+
+Socket units show the paths or ports on which they listen, and which service unit they activate.
+
+![Socket unit listening path](/images/services-socket-listen.png)
+
+Service units show their current memory usage and which socket unit activated them (if any). The static relationships like "Requires:" are now collapsed by default, to take less space and make it easier to see the journal output below.
+
+![Service memory usage](/images/services-mem-usage.png)
+
+## Developer API: Watch for file changes without reading
+
+You can use the [cockpit.file.watch() API](https://cockpit-project.org/guide/latest/cockpit-file.html#cockpit-file-notify) to update your Cockpit page when a file changes. So far this has always automatically read the file as well, and the called handler function returned the contents.
+
+This is impractical for very large files (such as databases) or other cases where the contents is unimportant. You can now set an option `read: false` to disable the automatic reading; then the callback's `content` will always be `null`.
```

Comment:
```suggestion
Reading files in impractical in many cases, as contents may be unimportant or a file could be large (which happens with databases). Therefore, the watch API has gained an option, `read: false`, to disable the automatic reading. When read is set to false, the callback's `content` will always be `null`.
```

Reply:
s/in/is/, and /which happens with/for example a/